pkg/server/incident.go

```diff
@@ -1,78 +1,370 @@
+	}
+
+	logger.Debug().Interface("request", request).Send()
+
+	incident, err := DbDef.IncidentFromAPI(&request)
+	if err != nil {
+		logger.Error().Err(err).Msg("error parsing request")
+
+		return echo.ErrInternalServerError
+	}
+
+	incidentUUID, err := uuid.Parse(incidentID)
+	if err != nil {
+		logger.Error().Err(err).Msg("error parsing id")
+
+		return echo.ErrInternalServerError
+	}
+
+	incident.ID = &incidentUUID
+
+	res := i.dbCon.Updates(&incident)
+	if res.Error != nil {
+		logger.Error().Err(err).Msg("error updating incident")
+
+		return echo.ErrInternalServerError
+	}
+
+	if res.RowsAffected == 0 {
+		logger.Warn().Msg("incident not found")
+
+		return echo.ErrNotFound
+	}
+
+	return ctx.NoContent(http.StatusNoContent) //nolint:wrapcheck
+}
+
+// GetIncidentUpdates retrieves a list of all updates for one incident.
+func (i *Implementation) GetIncidentUpdates(ctx echo.Context, incidentID api.IncidentIdPathParameter) error {
+	var incidentUpdates []DbDef.IncidentUpdate
+
+	logger := i.logger.With().Str("handler", "GetIncidentUpdates").Str("id", incidentID).Logger()
+	logger.Debug().Send()
+
+	res := i.dbCon.Where("incident_id = ?", incidentID).Find(&incidentUpdates)
+	if res.Error != nil {
+		logger.Error().Err(res.Error).Msg("error loading incident updates")
+
+		return echo.ErrInternalServerError
 	}
 
-	incidentList := make([]*api.Incident, len(incidents))
-	for incidentIndex := range incidentList {
-		incidentList[incidentIndex] = IncidentFromDB(incidents[incidentIndex])
+	data := make([]api.IncidentUpdateResponseData, len(incidentUpdates))
+	for incidentUpdateIndex, incidentUpdate := range incidentUpdates {
+		data[incidentUpdateIndex] = incidentUpdate.ToAPIResponse()
 	}
 
-	return ctx.JSON(http.StatusOK, incidentList)
+	return ctx.JSON(http.StatusOK, api.IncidentUpdateListResponse{ //nolint:wrapcheck
+		Data: data,
+	})
 }
 
-// IncidentFromDB is a helper function, converting a [db.Incident] to an [api.Incident].
-func IncidentFromDB(incident *DbDef.Incident) *api.Incident {
-	return &api.Incident{
-		Affects:     incident.GetAffectsIds(),
-		BeganAt:     incident.BeganAt,
-		Description: incident.Description,
-		EndedAt:     incident.EndedAt,
-		Id:          string(incident.ID),
-		ImpactType:  incident.ImpactType.Slug,
-		Phase:       incident.Phase.Slug,
-		Title:       incident.Title,
-		Updates:     IncidentUpdatesFromDB(incident.Updates),
+// CreateIncidentUpdate handles updates to an update for one incident.
+func (i *Implementation) CreateIncidentUpdate(ctx echo.Context, incidentID api.IncidentIdPathParameter) error {
+	var request api.CreateIncidentUpdateJSONRequestBody
+
+	logger := i.logger.With().Str("handler", "CreateIncidentUpdate").Str("id", incidentID).Logger()
+
+	err := ctx.Bind(&request)
+	if err != nil {
+		logger.Error().Err(err).Msg("error binding request")
+
+		return echo.ErrInternalServerError
+	}
+
+	order, err := DbDef.GetHighestIncidentUpdateOrder(i.dbCon, incidentID)
```

Comment:
Again, there is a writing `Create` after reading from the database. This needs a transaction, as well.